frontend/src/components/Forms/CoffeeChatsForm/CoffeeChatsForm.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useEffect, useState } from 'react';
+import { Form, Dropdown } from 'semantic-ui-react';
+import { Emitters } from '../../../utils';
+import { useSelf } from '../../Common/FirestoreDataProvider';
+import { MembersAPI } from '../../../API/MembersAPI';
+import CoffeeChatsDashboard from './CoffeeChatsDashboard';
+import styles from './CoffeeChatsForm.module.css';
+import CoffeeChatAPI from '../../../API/CoffeeChatAPI';
+
+const CoffeeChatsForm: React.FC = () => {
+  const userInfo = useSelf()!;
+  const [member, setMember] = useState<IdolMember | undefined>(undefined);
+  const [category, setCategory] = useState('');
+  const [membersList, setMembersList] = useState<IdolMember[]>([]);
+  const [approvedChats, setApprovedChats] = useState<CoffeeChat[]>([]);
+  const [pendingChats, setPendingChats] = useState<CoffeeChat[]>([]);
+  const [rejectedChats, setRejectedChats] = useState<CoffeeChat[]>([]);
+  const [isChatLoading, setIsChatLoading] = useState<boolean>(true);
+  const [slackLink, setSlackLink] = useState<string>('');
+
+  useEffect(() => {
+    MembersAPI.getAllMembers().then((members) => setMembersList(members));
+    CoffeeChatAPI.getCoffeeChatsByUser(userInfo).then((coffeeChat) => {
+      setApprovedChats(coffeeChat.filter((chat) => chat.status === 'approved'));
+      setPendingChats(coffeeChat.filter((chat) => chat.status === 'pending'));
+      setRejectedChats(coffeeChat.filter((chat) => chat.status === 'rejected'));
+      setIsChatLoading(false);
+    });
+  }, [userInfo]);
+
+  const coffeeChatExists = async (): Promise<boolean> => {
+    const chats = await CoffeeChatAPI.getCoffeeChatsByUser(userInfo);
+    return chats.some((chat) => chat.otherMember.netid === member?.netid);
+  };
+
+  const submitCoffeeChat = async () => {
+    if (!member) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Team Member Selected',
+        contentMsg: 'Please select a team member!'
+      });
+    } else if (!slackLink) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Image Entered',
+        contentMsg: 'Please enter an image link!'
+      });
+    } else if (!category) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Category Entered',
+        contentMsg: 'Please enter a category!'
+      });
+    } else if (member.netid === userInfo.netid) {
+      Emitters.generalError.emit({
+        headerMsg: 'Cannot Coffee Chat Yourself',
+        contentMsg: 'Please submit a coffee chat with another member!'
+      });
+    } else if (await coffeeChatExists()) {
```

Comment:
How about instead of making a request to do a verification here, we just verify with the frontend state? Like, check the existing list of coffee chats that have been loaded in in `pendingChats` and `approvedChats`. This will make the submission slightly faster (since we're not making another API call), and worst case, the backend will deny the request if the frontend is wrong somehow.

Removing this also lets you get rid of the `await/async` usage from here and where you're calling `submitCoffeeChat` down below.